perfkitbenchmarker/spark_service.py

```diff
@@ -87,14 +103,24 @@ def __init__(self, spark_service_spec):
     is_user_managed = spark_service_spec.static_cluster_id is not None
     super(BaseSparkService, self).__init__(user_managed=is_user_managed)
     self.spec = spark_service_spec
+    # If only the worker group is specified, assume the master group is
+    # configured the same way.
+    if spark_service_spec.master_group is None:
+      self.spec.master_group = copy.copy(self.spec.worker_group)
+      self.spec.master_group.vm_count = 1
     self.cluster_id = spark_service_spec.static_cluster_id
-    self.num_workers = spark_service_spec.num_workers
-    self.machine_type = spark_service_spec.machine_type
-    self.project = spark_service_spec.project
+    if spark_service_spec.master_group.vm_spec.zone:
+      self.zone = spark_service_spec.master_group.vm_spec.zone
+    elif FLAGS.zones:
+      self.zone = FLAGS.zones[0]
+    else:
+      self.zone = None
+    self.project = FLAGS.project
```

Comment:
We already talked about this, but GceVmSpecs should have this so you can grab it from the either the worker or master spec